packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -594,6 +594,12 @@ function transform(codeGen: CodeGen): t.Expression {
             data.push(codeGen.genRef(ref));
         }
 
+        // Spread
+        if (spread) {
```

Comment:
Question on the ordering here, should an `innerHTML` coming in through `lwc:spread` override the `lwc:inner-html` directive?

Reply:
Good question! It should override that HTML just like it overrides all other properties. Added a test for it.

Reply:
I see we have a test for spread overriding props already:

https://github.com/salesforce/lwc/blob/062d08c0a9ea41742cb3db37ce809aca1967d6c9/packages/%40lwc/integration-karma/test/spread/x/test/test.html#L3

However, this tests `onclick`, which I think behaves a bit differently in the compiler compared to standard props. Can we add a test to ensure that spread overrides explicit props? (If it doesn't already exist.)

Reply:
`name` in the above example is a regular prop.

Reply:
Ah, my mistake! Totally missed that.